### DISCUSSION TITLE
Error with fragment picking

### DISCUSSION THREAD

**Discussion by wendi-luo** [RosettaCommons/rosetta#223] [Rosetta] [answered]
Hello everyone,

I'm very new to using Rosetta and I'm having problems generating mers. I've followed the tutorials and the documentation, but I continue to receive an error.

I've created a directory (input_files) with the following files: a FASTA sequence, an ss2 file (generated using PsiPred), and a PDB file (generated by AlphaFold). 

This is the command I use in the shell:
`fragment_picker.static.linuxgccrelease @frag_flags`

Before this step, I have done repack to protein by
`FlexPepDocking.macosclangrelease -s com.pdb -flexpep_prepack -ex1 -ex2aro -use_input_sc -nstruct 1`

The error showed as:
![image](https://github.com/user-attachments/assets/c7437599-333a-4f70-ae24-00903e872919)

Every file I used were compressed in 10.1.zip which I upload in attatch.
Can you please help me making this fragment picker work?
Thanks a lot!
[10.1.zip](https://github.com/user-attachments/files/17520658/10.1.zip)

**Comment by dgront** [comment]
Please check if the length of your sequence in all your input files is the
same, i.e. whether the protein from PsiPred has the same number of amino
acids as your FASTA.

Also, your query sequence might be too short, either don't pick 9-mers or
make it a bit shorter (like 5-mers).
Simply check, if it crashes on 3-mers only.


Best regards,
Dominik Gront

pt., 25 paź 2024 o 12:37 wendi-luo ***@***.***> napisał(a):

**Comment by roccomoretti** [comment, marked as answer]
It looks like your input files have DOS/Windows line endings ('\r\n'). The issue is that when Rosetta reads in your fasta sequence, it expects Unix-style line endings ('\n'). This means that you have an "extra" '\r' character in your query sequence. As such, it runs this check and sees that 11 is not equal to 10 and you get the error message you see.

The fix is to run the `dos2unix` program over your input files. This will convert the line endings to their proper form for a Mac/Linux machine, and you won't get spurious '\r' characters messing things up.

**Reply by wendi-luo** [reply]
That's worked! Many thanks!

**Comment by roccomoretti** [comment]
Note that 9-mers are indeed probably overkill for a 10-mer peptide. For peptides that short, FlexPepDock usually makes do with 3-mers and 5-mers.

That said, once you correct the line ending issue, you should be able to generate 9-mer fragments (whether or not they're practically useful).
